modules/sharedaddy/sharing-service.php

```diff
@@ -170,8 +170,14 @@ public function get_blog_services() {
 		$enabled  = get_option( 'sharing-services' );
 		$services = $this->get_all_services();
 
-		if ( !is_array( $options ) )
+		/**
+		 * Check if options exist and are well formatted.
+		 * This avoids issues on sites with corrupted options.
+		 * @see https://github.com/Automattic/jetpack/issues/6121
+		 */
+		if ( ! is_array( $options ) || ! isset( $options['button_style'] ) ) {
```

Comment:
Should this do `isset( $options['button_style'],  $options['global'] )` to test for `global` as well?  Just as we're about to use it in the subsequent line after the conditional.